tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb

```diff
@@ -0,0 +1,801 @@
+    "    \n",
+    "REGEX_MAPPINGS = create_regex_dictionaries(REGEX_CONF_PATH)\n",
+    "\n",
+    "parts = [dask.delayed(pipeline)(x, EVENT_CODES_OF_INTEREST, REGEX_MAPPINGS) for x in logs_gddf.to_delayed()]\n",
+    "temp_df = dask_cudf.from_delayed(parts)\n",
+    "# Bring data back to a single GPU, for downstream graph analytics\n",
+    "auth_gdf = temp_df.compute()\n",
+    "print(auth_gdf.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Read edge definitions from JSON file\n",
+    "\n",
+    "Now the parsing of Windows Event Logs has concluded, next we prepare for the network mapping portion. Within the edge definitions configuration file we define our edges by indicating the source and destination for each edge, referencing the column names of our input data.  \n",
+    "  \n",
+    "Below we also read in the netflow data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filename = 'conf/network_mapping_configs/lanl-edge-definitions.json'\n",
+    "with open(filename) as f:\n",
+    "    edge_defs = json.load(f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Build network mapping edge list\n",
+    "\n",
+    "This function helps to recognize the data types we read in via csv, particulary the string objects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_dtypes(fn, delim, floats, strings):\n",
+    "    with open(fn, errors='replace') as fp:\n",
+    "        header = fp.readline().strip()\n",
+    "    \n",
+    "    types = []\n",
+    "    for col in header.split(delim):\n",
+    "        if 'date' in col:\n",
+    "            types.append((col, 'date'))\n",
+    "        elif col in floats:\n",
+    "            types.append((col, 'float64'))\n",
+    "        elif col in strings:\n",
+    "            types.append((col, 'str'))\n",
+    "        else:\n",
+    "            types.append((col, 'int32'))\n",
+    "\n",
+    "    return OrderedDict(types)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Read in Netflow Data\n",
+    "\n",
+    "The netflow data is also provided by [Los Alamos National Laboratory](https://csr.lanl.gov/data/2017.html)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p data/input/lanl\n",
+    "!if [ ! -f data/input/lanl/netflow.csv ]; then tar -xzvf /data/lanl/netflow.tar.gz -C data/input/lanl; fi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "flow_input_path = 'data/input/lanl/netflow.csv'\n",
+    "dtypes_data_processed = get_dtypes(flow_input_path, ',', floats=[], strings=[\"SrcDevice\", \"DstDevice\"])   \n",
```

Comment:
This is fine, but I'd be interested in seeing if cudf in 0.7 can correctly infer the column types. If so, that would be great, as at least in my opinion I find using this function a bit annoying 😄